src/com/epam/javatrainings/classdecomposition/pizzeria_palmetto/model/Pizza.java

```diff
@@ -0,0 +1,62 @@
+package com.epam.javatrainings.classdecomposition.pizzeria_palmetto.model;
+
+import com.epam.javatrainings.classdecomposition.pizzeria_palmetto.services.Constants;
+
+import static com.epam.javatrainings.classdecomposition.pizzeria_palmetto.services.Constants.*;
+import java.util.ArrayList;
+import java.util.List;
+
+public class Pizza {
+  private String name;
+  private Type type;
+  private int quantity;
+  public static List<String> ingredientsList;
+
+  public Pizza(String name, String type, int quantity) {
+    this.setQuantity(quantity);
+    this.name = name;
+    this.setType(type);
+    ingredientsList = new ArrayList<>();
+  }
+
+  public String getName() {
+    return name;
+  }
+
+  public Type getType() {
+    return type;
+  }
+
+  public int getQuantity() {
+    return quantity;
+  }
+
+  public void setQuantity(int quantity) {
+    if (quantity > QUANTITY_PIZZA_MIN && quantity <= QUANTITY_PIZZA_MAX) {
+      this.quantity = quantity;
+    } else if (quantity > QUANTITY_PIZZA_MAX) {
+      this.quantity = QUANTITY_PIZZA_MAX;
+    } else if (quantity <= QUANTITY_PIZZA_MIN) {
```

Comment:
if these two logical expressions are false the third will always true,
so you write : 
else this.quantity = QUANTITY_PIZZA_DEFAULT;

Reply:
done